internal/server/dataset.go

```diff
@@ -279,53 +279,43 @@ func (ds *Dataset) StoreEntities(entities []*Entity) (Error error) {
 		// assume different from a previous version
 		isDifferent := true
 
-		if !isnew {
-			opts1 := badger.DefaultIteratorOptions
-			opts1.PrefetchValues = false
-			opts1.Reverse = true
-			entityLocatorIterator := rtxn.NewIterator(opts1)
-			entityLocatorPrefixBuffer := make([]byte, 14)
-			copy(entityLocatorPrefixBuffer, entityIdBuffer[:14]) // fixme: optimise a bit with some buffer reuse.
-			entityLocatorPrefixBuffer = append(entityLocatorPrefixBuffer, 0xFF)
-
-			countAsNew := true
-
-			for entityLocatorIterator.Seek(entityLocatorPrefixBuffer); entityLocatorIterator.ValidForPrefix(entityLocatorPrefixBuffer[:14]); {
-				item := entityLocatorIterator.Item()
-				k := item.Key()
-				countAsNew = false // if it enters here, we find the existing item, and dont count it
-				existingEntityDataLength := binary.BigEndian.Uint16(k[22:])
-				if existingEntityDataLength == uint16(jsonLength) {
-					// load data as length is the same
-					err := item.Value(func(val []byte) error {
-						_ = ds.store.statsdClient.Count("ds.read.bytes", int64(len(val)), tags, 1)
-						existingEntity := &Entity{}
-						err := json.Unmarshal(val, existingEntity)
-						if err != nil {
-							return err
-						}
-
-						if reflect.DeepEqual(existingEntity.References, e.References) && reflect.DeepEqual(existingEntity.Properties, e.Properties) {
-							isDifferent = false
-						} else {
-							isDifferent = true
-						}
-						return nil
-					})
-
-					if err != nil {
-						return err
-					}
-				} else {
-					isDifferent = true
-				}
+		datasetEntitiesLatestVersionKey := make([]byte, 14)
+		binary.BigEndian.PutUint16(datasetEntitiesLatestVersionKey, DATASET_LATEST_ENTITIES)
+		binary.BigEndian.PutUint32(datasetEntitiesLatestVersionKey[2:], ds.InternalID)
+		binary.BigEndian.PutUint64(datasetEntitiesLatestVersionKey[6:], rid)
+		// fixme: optimise to have txntime in key and not need a value
 
-				// only need the first one we find
-				break
+		var previousEntityIdBuffer []byte
+		var prevEntity *Entity
+		if previousEntityIdBufferValue, err := rtxn.Get(datasetEntitiesLatestVersionKey); err == nil {
+			previousEntityIdBuffer, err = previousEntityIdBufferValue.ValueCopy(nil)
```

Comment:
@rompetroll in this section - compared with before - i think we have lost the size comparison check. This is important when there isnt a change as it avoids the parsing.